jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -159,5 +168,101 @@ public void removeComment(final String serverTemplateId, final String containerI
         return comparing(CaseMilestoneSummary::getStatus).thenComparing(request.getSortByAsc() ? comparatorByName: comparatorByName.reversed());
     }
 
+    @Override
+    public Actions getCaseActions(String serverTemplateId, String container, String caseId, String userId) {
+        List<CaseActionSummary> allActionsSummaries = getAllUserTaskActionSummaries(container, caseId, userId);
+
+        List<CaseActionSummary> availableActions = getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.AVAILABLE);
+        availableActions.addAll(getAdHocActions(serverTemplateId, container, caseId));
+        Actions actions = new Actions();
+        actions.setAvailableActions(availableActions);
+        actions.setInProgressAction(getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.IN_PROGRESS));
+        actions.setCompleteActions(getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.COMPLETED_FAIL, CaseActionStatus.COMPLETED_SUCCESS));
+        return actions;
+    }
+
+    public List<CaseActionSummary> getAllUserTaskActionSummaries(String containerId, String caseId, String userId) {
+        List<CaseActionSummary> allActionsSummaries = new ArrayList<>();
+
+        List<String> allStatus = new ArrayList<>();
+        allStatus.addAll(CaseActionStatus.AVAILABLE.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.IN_PROGRESS.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_FAIL.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_SUCCESS.getTaskStatuses());
+
+        List<NodeInstance> activeNodes = client.getActiveNodes(containerId, caseId, 0, PAGE_SIZE_UNLIMITED);
+        List<CaseActionSummary> activeNoUserTaskActionsSummaries =
+                activeNodes.stream()
+                        .filter(c -> !Arrays.asList(CaseActionNodeInstanceMapper.NODE_TYPE_HUMAN_TASK_NODE,
+                                CaseActionNodeInstanceMapper.NODE_TYPE_Dynamic_HUMAN_TASK)
+                                .contains(c.getNodeType()))
+                        .map(new CaseActionNodeInstanceMapper())
+                        .map(s -> {
+                            s.setStatus(CaseActionStatus.IN_PROGRESS.getTaskStatuses().get(0));
+                            return s;
+                        }).collect(toList());
+
+        if (!activeNoUserTaskActionsSummaries.isEmpty()) {
+            allActionsSummaries.addAll(activeNoUserTaskActionsSummaries);
+        }
+
+        List<TaskSummary> actionHumanTaskSummaries = client.findCaseTasksAssignedAsBusinessAdministrator(caseId, userId, allStatus, 0, PAGE_SIZE_UNLIMITED, "createdOn", false);
```

Comment:
@cristianonicolai I added the  filter:
.filter(c -> !Arrays.asList(CaseActionNodeInstanceMapper.NODE_TYPE_HUMAN_TASK_NODE,
                                CaseActionNodeInstanceMapper.NODE_TYPE_Dynamic_HUMAN_TASK)
 over the active nodes to avoid create "inprogess" actions over the human tasks than comes in activeNodes. That are retrieved by client.findCaseTasksAssignedAsBusinessAdministrator that only check the current user is designed a bussinessAdministrator.